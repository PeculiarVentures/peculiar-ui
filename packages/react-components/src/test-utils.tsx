import React from 'react';
import {
  render,
  screen,
  fireEvent,
  waitFor,
  renderHook,
  act,
} from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { ThemeProvider } from './styles/theme_provider';

const renderWithWrapper = (
  ui: React.ReactElement,
) => (
  render(
    ui,
    {
      wrapper: ThemeProvider,
    },
  )
);

export {
  userEvent,
  fireEvent,
  renderWithWrapper,
  screen,
  waitFor,
  renderHook,
  act,
};
