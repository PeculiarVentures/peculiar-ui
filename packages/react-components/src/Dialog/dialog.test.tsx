import React from 'react';
import { renderWithWrapper as render } from '../test-utils';
import { Dialog } from './index';

describe('<Dialog />', () => {
  it('should render with default styles', () => {
    const { baseElement } = render(
      <Dialog open={false}>
        Dialog Content
      </Dialog>,
    );

    expect(baseElement).toMatchSnapshot();
  });

  it('should render with default open styles', () => {
    const { baseElement } = render(
      <Dialog open>
        Dialog Content
      </Dialog>,
    );

    expect(baseElement).toMatchSnapshot();
  });

  it('should render fullscreen', () => {
    const { baseElement } = render(
      <Dialog open fullScreen>
        Dialog Content
      </Dialog>,
    );

    expect(baseElement).toMatchSnapshot();
  });

  describe('sizes', () => {
    const sizes: Array<React.ComponentProps<typeof Dialog>['size']> = [
      'small',
      'medium',
      'large',
    ];

    sizes.forEach((size) => {
      it(`size "${size}"`, () => {
        const { baseElement } = render(
          <Dialog open size={size}>
            Dialog Content
          </Dialog>,
        );

        expect(baseElement).toMatchSnapshot();
      });
    });
  });
});
