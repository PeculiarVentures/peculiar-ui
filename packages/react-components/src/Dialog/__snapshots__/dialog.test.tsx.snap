// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Dialog /> should pass className 1`] = `
.emotion-0 {
  position: fixed;
  z-index: 1300;
  top: 0;
  bottom: 0;
  right: 0;
  left: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
}

.emotion-1 {
  background: var(--pv-color-black);
  z-index: -1;
  position: fixed;
  right: 0;
  bottom: 0;
  top: 0;
  left: 0;
  -webkit-tap-highlight-color: transparent;
}

.emotion-2 {
  border-radius: 4px;
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  max-height: calc(100% - 60px);
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  margin: var(--pv-size-base-6);
  position: relative;
  outline: none;
  overflow: hidden;
  border: 0;
  padding: 0;
  box-shadow: var(--pv-shadow-dark-hight);
  max-width: 640px;
  color: var(--pv-color-black);
  background-color: var(--pv-color-white);
}

<body>
  <div />
  <div
    aria-hidden="false"
    class="emotion-0"
    role="presentation"
  >
    <div
      aria-hidden="true"
      class="emotion-1"
      style="opacity: 0.5; transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
    />
    <fieldset
      class="my-class-name emotion-2"
      role="dialog"
      style="opacity: 1; transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
      tabindex="-1"
    >
      Dialog Content
    </fieldset>
  </div>
</body>
`;

exports[`<Dialog /> should render fullscreen 1`] = `
.emotion-0 {
  position: fixed;
  z-index: 1300;
  top: 0;
  bottom: 0;
  right: 0;
  left: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
}

.emotion-1 {
  background: var(--pv-color-black);
  z-index: -1;
  position: fixed;
  right: 0;
  bottom: 0;
  top: 0;
  left: 0;
  -webkit-tap-highlight-color: transparent;
  background-color: transparent;
}

.emotion-2 {
  border-radius: 0;
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  max-height: none;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  margin: 0;
  position: relative;
  outline: none;
  overflow: hidden;
  border: 0;
  padding: 0;
  box-shadow: var(--pv-shadow-dark-hight);
  max-width: 100%;
  height: 100%;
  color: var(--pv-color-black);
  background-color: var(--pv-color-white);
}

<body>
  <div />
  <div
    aria-hidden="false"
    class="emotion-0"
    role="presentation"
  >
    <div
      aria-hidden="true"
      class="emotion-1"
      style="opacity: 0.5; transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
    />
    <fieldset
      class="emotion-2"
      role="dialog"
      style="opacity: 1; transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
      tabindex="-1"
    >
      Dialog Content
    </fieldset>
  </div>
</body>
`;

exports[`<Dialog /> should render with default open styles 1`] = `
.emotion-0 {
  position: fixed;
  z-index: 1300;
  top: 0;
  bottom: 0;
  right: 0;
  left: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
}

.emotion-1 {
  background: var(--pv-color-black);
  z-index: -1;
  position: fixed;
  right: 0;
  bottom: 0;
  top: 0;
  left: 0;
  -webkit-tap-highlight-color: transparent;
}

.emotion-2 {
  border-radius: 4px;
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  max-height: calc(100% - 60px);
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  margin: var(--pv-size-base-6);
  position: relative;
  outline: none;
  overflow: hidden;
  border: 0;
  padding: 0;
  box-shadow: var(--pv-shadow-dark-hight);
  max-width: 640px;
  color: var(--pv-color-black);
  background-color: var(--pv-color-white);
}

<body>
  <div />
  <div
    aria-hidden="false"
    class="emotion-0"
    role="presentation"
  >
    <div
      aria-hidden="true"
      class="emotion-1"
      style="opacity: 0.5; transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
    />
    <fieldset
      class="emotion-2"
      role="dialog"
      style="opacity: 1; transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
      tabindex="-1"
    >
      Dialog Content
    </fieldset>
  </div>
</body>
`;

exports[`<Dialog /> should render with default styles 1`] = `
<body>
  <div />
</body>
`;

exports[`<Dialog /> sizes size "large" 1`] = `
.emotion-0 {
  position: fixed;
  z-index: 1300;
  top: 0;
  bottom: 0;
  right: 0;
  left: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
}

.emotion-1 {
  background: var(--pv-color-black);
  z-index: -1;
  position: fixed;
  right: 0;
  bottom: 0;
  top: 0;
  left: 0;
  -webkit-tap-highlight-color: transparent;
}

.emotion-2 {
  border-radius: 4px;
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  max-height: calc(100% - 60px);
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  margin: var(--pv-size-base-6);
  position: relative;
  outline: none;
  overflow: hidden;
  border: 0;
  padding: 0;
  box-shadow: var(--pv-shadow-dark-hight);
  max-width: 1024px;
  color: var(--pv-color-black);
  background-color: var(--pv-color-white);
}

<body>
  <div />
  <div
    aria-hidden="false"
    class="emotion-0"
    role="presentation"
  >
    <div
      aria-hidden="true"
      class="emotion-1"
      style="opacity: 0.5; transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
    />
    <fieldset
      class="emotion-2"
      role="dialog"
      style="opacity: 1; transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
      tabindex="-1"
    >
      Dialog Content
    </fieldset>
  </div>
</body>
`;

exports[`<Dialog /> sizes size "medium" 1`] = `
.emotion-0 {
  position: fixed;
  z-index: 1300;
  top: 0;
  bottom: 0;
  right: 0;
  left: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
}

.emotion-1 {
  background: var(--pv-color-black);
  z-index: -1;
  position: fixed;
  right: 0;
  bottom: 0;
  top: 0;
  left: 0;
  -webkit-tap-highlight-color: transparent;
}

.emotion-2 {
  border-radius: 4px;
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  max-height: calc(100% - 60px);
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  margin: var(--pv-size-base-6);
  position: relative;
  outline: none;
  overflow: hidden;
  border: 0;
  padding: 0;
  box-shadow: var(--pv-shadow-dark-hight);
  max-width: 640px;
  color: var(--pv-color-black);
  background-color: var(--pv-color-white);
}

<body>
  <div />
  <div
    aria-hidden="false"
    class="emotion-0"
    role="presentation"
  >
    <div
      aria-hidden="true"
      class="emotion-1"
      style="opacity: 0.5; transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
    />
    <fieldset
      class="emotion-2"
      role="dialog"
      style="opacity: 1; transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
      tabindex="-1"
    >
      Dialog Content
    </fieldset>
  </div>
</body>
`;

exports[`<Dialog /> sizes size "small" 1`] = `
.emotion-0 {
  position: fixed;
  z-index: 1300;
  top: 0;
  bottom: 0;
  right: 0;
  left: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
}

.emotion-1 {
  background: var(--pv-color-black);
  z-index: -1;
  position: fixed;
  right: 0;
  bottom: 0;
  top: 0;
  left: 0;
  -webkit-tap-highlight-color: transparent;
}

.emotion-2 {
  border-radius: 4px;
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  max-height: calc(100% - 60px);
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  margin: var(--pv-size-base-6);
  position: relative;
  outline: none;
  overflow: hidden;
  border: 0;
  padding: 0;
  box-shadow: var(--pv-shadow-dark-hight);
  max-width: 310px;
  color: var(--pv-color-black);
  background-color: var(--pv-color-white);
}

<body>
  <div />
  <div
    aria-hidden="false"
    class="emotion-0"
    role="presentation"
  >
    <div
      aria-hidden="true"
      class="emotion-1"
      style="opacity: 0.5; transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
    />
    <fieldset
      class="emotion-2"
      role="dialog"
      style="opacity: 1; transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
      tabindex="-1"
    >
      Dialog Content
    </fieldset>
  </div>
</body>
`;
